data/txHandler.go

```diff
@@ -282,41 +309,135 @@ func (handler *TxHandler) asyncVerifySignature(arg interface{}) interface{} {
 		// we failed to write to the output queue, since the queue was full.
 		// adding the metric here allows us to monitor how frequently it happens.
 		transactionMessagesDroppedFromPool.Inc(nil)
+
+		// delete from duplicate caches to give it chance to be re-submitted
+		// this relatively rare operation and implementation is expensive (requires re-hashing)
+		handler.deleteFromCaches(tx.rawmsg.Data, tx.unverifiedTxGroup)
+
 	}
 	return nil
 }
 
+func (handler *TxHandler) deleteFromCaches(msg []byte, unverifiedTxGroup []transactions.SignedTxn) {
+	var result []byte
+	for i := range unverifiedTxGroup {
```

Comment:
This operation is performed at the worst possible time (when the system is overloaded and dropping txns). 
Why not piggyback the hash to the  `txBacklogMsg` so that it can be reused here, instead of recalculating it?

Reply:
good point